pkg/blockdevice/v1alpha1/blockdevice.go

```diff
@@ -0,0 +1,242 @@
+func (bd *BlockDevice) Build() (*ndm.BlockDevice, []error) {
+	bd.validate()
+	if len(bd.errs) > 0 {
+		return nil, bd.errs
+	}
+	return bd.BlockDevice, nil
+}
+
+// validate validates the block device object against the check predicates.
+func (bd *BlockDevice) validate() {
+	for _, c := range checkPredicatesFuncs {
+		msg, ok := c(bd)
+		if !ok {
+			bd.errs = append(bd.errs, predicateFailedError(msg))
+		}
+	}
+}
+
+// WithName method fills the name field of BlockDevice object.
+func (bd *BlockDevice) WithName(name string) *BlockDevice {
+	WithName(name)(bd)
+	return bd
+}
+
+// WithName function is used by WithName method as a util.
+// Ideas is to give flexibility for building object by using dot operator as well as passing
+// build predicated to the New constructor.
+func WithName(name string) BuildOptionFunc {
+	return func(bd *BlockDevice) {
+		bd.BlockDevice.Name = name
+	}
+}
+
+// WithState method fills the name field of BlockDevice object.
+func (bd *BlockDevice) WithState(state string) *BlockDevice {
+	WithState(state)(bd)
+	return bd
+}
+
+// WithState function is used by WithState method as a util.
+// Ideas is to give flexibility for building object by using dot operator as well as passing
+// build predicated to the New constructor.
+func WithState(state string) BuildOptionFunc {
+	return func(bd *BlockDevice) {
+		bd.BlockDevice.Status.State = state
+	}
+}
+
+//checkName validate the name field of BlockDevice object.
+func checkName(db *BlockDevice) (string, bool) {
+	if db.BlockDevice.Name == "" {
+		//TODO: Think about having some good organization in putting error messages.
+		return "blockDevice name field on the object may not be empty", false
+	}
+	return "", true
+}
+
+// Filter adds filters on which the blockdevice has to be filtered
+func (bdl *BlockDeviceList) Filter(predicateKeys ...string) *BlockDeviceList {
+	// Initialize filtered block device list
+	filteredBlockDeviceList := &BlockDeviceList{
+		BlockDeviceList: &ndm.BlockDeviceList{},
+		errs:            nil,
+	}
+	errMsg, ok := bdl.Hasitems()
+	if !ok {
+		filteredBlockDeviceList.errs = append(filteredBlockDeviceList.errs, errors.New(errMsg))
+		return filteredBlockDeviceList
+	}
+	filteredBlockDeviceList = bdl
+	for _, key := range predicateKeys {
+		filteredBlockDeviceList = filterOptionFuncMap[key](filteredBlockDeviceList)
+	}
+	return filteredBlockDeviceList
+}
+
+//filterInactive filter and give out all the inactive block device
+func filterInactive(orignialList *BlockDeviceList) *BlockDeviceList {
+	filteredList := &BlockDeviceList{
+		BlockDeviceList: &ndm.BlockDeviceList{},
+		errs:            nil,
+	}
+	for _, device := range orignialList.Items {
+		if device.Status.State == "Inactive" {
+			filteredList.Items = append(filteredList.Items, device)
+		}
+	}
+	return filteredList
+}
+
+//filterInactiveReverse give out all the block device except inactive block devices
+func filterInactiveReverse(orignialList *BlockDeviceList) *BlockDeviceList {
```

Comment:
better name would be required like `filterNonInactive`

Reply:
Done